public/app/features/explore/Logs/LogsTable.tsx

```diff
@@ -74,36 +92,45 @@ export const LogsTable: React.FunctionComponent<Props> = (props) => {
         field.config = {
           ...field.config,
           custom: {
-            filterable: true,
             inspect: true,
+            filterable: true, // This sets the columns to be filterable
             ...field.config.custom,
           },
+          // This sets the individual field value as filterable
+          filterable: isFieldFilterable(field, logsFrame ?? undefined),
         };
       }
 
       return frameWithOverrides;
     },
-    [logsSortOrder, range, splitOpen, timeZone]
+    // We dont want to re-render whenever the splitOpen function changes, so it's being purposefully excluded from the deps
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [logsSortOrder, timeZone, props.datasourceType, range]
   );
 
+  /**
+   * Known issue here is that there is a re-render of the table when the set of labels has changed,
+   * the transformations are added after the fresh data has already been rendered
+   */
   useEffect(() => {
     const prepare = async () => {
-      if (!logsFrames || !logsFrames.length) {
+      if (!logFrameRaw || !logsFrame) {
         setTableFrame(undefined);
         return;
       }
-      // TODO: This does not work with multiple logs queries for now, as we currently only support one logs frame.
-      let dataFrame = logsFrames[0];
 
-      const logsFrame = parseLogsFrame(dataFrame);
+      // Tables currently only support one frame
+      let dataFrame = logFrameRaw;
+
       const timeIndex = logsFrame?.timeField.index;
       dataFrame = sortDataFrame(dataFrame, timeIndex, logsSortOrder === LogsSortOrder.Descending);
 
       // create extract JSON transformation for every field that is `json.RawMessage`
       // TODO: explore if `logsFrame.ts` can help us with getting the right fields
+      // TODO Why is typeInfo not defined on the Field interface?
```

Comment:
I couldn't really find any documentation around the Field.typeInfo property in the frontend, it's only present for loki? Should probably ask @svennergr or someone about this before merge.

Reply:
Ideally there is a less hacky solution that doesn't require the dataSource type